python/cuml/metrics/hinge_loss.pyx

```diff
@@ -0,0 +1,120 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cupy as cp
+from cuml.preprocessing import LabelEncoder, LabelBinarizer
+import cudf
+
+
+def cython_hinge_loss(y_true, pred_decision, labels=None, sample_weights=None):
+    """
+    Calculates non-regularized hinge loss. Adapted from scikit-learn hinge loss
+
+    Parameters
+    ----------
+    y_true: cuDF Series or cuPy array of shape (n_samples,)
+            True labels, consisting of labels for the classes.
+            In binary classification, the positive label must be
+            greater than negative class
+
+    pred_decision: cuDF DataFrame or cuPy array of shape (n_samples,) or
+                   (n_samples, n_classes)
+                   Predicted decisions, as output by decision_function (floats)
+
+    labels: cuDF Series or cuPy array, default=None
+            In multiclass problems, this must include all class labels.
+
+    sample_weight: cupy array of shape (n_samples,), default=None
+                   Sample weights to be used for computing the average
+
+    Returns
+    -------
+    loss : float.
+           The average hinge loss.
+    """
+
+    # Check types of the inputs
+    if not hasattr(y_true, "__cuda_array_interface__"):
```

Comment:
Could we avoid some of this error-checking by using `input_to_cuml_array` to handle any necessary conversions?

Reply:
I haven't used that for error-checking before, I can try that out and see if it works.

Reply:
cuml array is not accept by LabelEncoder at the moment, so there needs to be a few cyclic conversions to accommodate that. It's probably better to have these checks for now and change it later, if needed. What do you think, @wphicks?

Reply:
Sounds great to me! These checks seem like the sort of thing we'll do over and over in the codebase, so we should probably have some utility function for it, but no need to hold up this PR on that basis.